### ISSUE TITLE
Bug: (Failed to compile. [eslint] Cannot read properties of undefined (reading 'getTokensBetween') Occurred while linting /home/itpmbot-dev/builds/K/0/itpm-chat-bot/bot-frontend/src/index.jsx:2 Rule: "perfectionist/sort-imports")

### ISSUE THREAD

**Issue by evgeniyworkbel** [azat-io/eslint-plugin-perfectionist#101]
### Describe the bug

After upgrading plugin from 2.4.0 to 2.5.0 our app build have failed. Why?

```
Failed to compile.
[eslint] Cannot read properties of undefined (reading 'getTokensBetween')
Occurred while linting /home/itpmbot-dev/builds/K1xYuVj3R/0/itpm-chat-bot/bot-frontend/src/index.jsx:2
Rule: "perfectionist/sort-imports"
```

### Code example

```
---
root: true

env:
  browser: true
  es2022: true

parserOptions:
  ecmaVersion: latest
  sourceType: module

plugins:
  - import
  - jsx-a11y
  - react
  - react-hooks
  - perfectionist

extends:
  - airbnb
  - plugin:import/recommended
  - plugin:react/recommended
  - plugin:react-hooks/recommended

rules:
  # base config rules
  arrow-body-style: "off"
  indent: ["error", 4, { "SwitchCase": 1 }]
  jsx-quotes: ["error", "prefer-single"]
  linebreak-style: "warn"
  lines-between-class-members: ["error", "always", { "exceptAfterSingleLine": true }]
  max-len: ["error", { "code": 135, "ignoreStrings": true, "ignoreComments": true }]
  no-console: "off" # replace with ["error", { "allow": ["warn", "error"] }] after major release
  no-param-reassign: ["error", { "props": false }]
  no-underscore-dangle: ["error", { "allow": ["_retry", ""] }]
  no-unused-vars: ["error", { "argsIgnorePattern": "^_" }]
  object-curly-newline: ["error", { "minProperties": 9, "consistent": true }]

  # plugin rules
  import/extensions: "off"
  import/no-extraneous-dependencies: ["error", {"packageDir": './'}]
  import/prefer-default-export: "off"
  import/no-default-export: "error"
  jsx-a11y/label-has-associated-control: ["error", { "assert": "either" }]
  react/function-component-definition: "off"
  react/jsx-filename-extension: ["warn", { "extensions": [ ".js", ".jsx" ] }]
  react/jsx-indent: ["error", 4, { "indentLogicalExpressions": true }]
  react/jsx-indent-props: ["off", 4]
  react/react-in-jsx-scope: "off"
  react/jsx-sort-props: ["error", { "callbacksLast": true, "shorthandLast": true, "reservedFirst": true }]
  react/jsx-props-no-spreading: "off"
  react/jsx-uses-react: "off"
  react/jsx-uses-vars: "error"
  react/prop-types: "off"
  react/prefer-exact-props: "off"
  perfectionist/sort-imports: [
      "error",
      {
        "groups": [
          "react",
          ["builtin", "external"],
          "internal-type",
          "internal",
          ["parent-type", "sibling-type", "index-type"],
          ["parent", "sibling", "index"],
          "side-effect",
          "side-effect-style",
          "style",
          "object",
          "unknown"
        ],
        "custom-groups": {
          "value": {
            "react": ["react", "react-*"],
          },
        },
      }
    ]

overrides:
  # by default, linter checks only .js file. It allows checking .jsx files in addition
  - files: ["*.jsx", "*.js"]  
  - files: ["**/reused/Icons/**/*.jsx"]
    rules:
      react/jsx-sort-props: "off"

settings:
  react:
    createClass: createReactClass
    pragma: React
    fragment: Fragment
    version: detect
    flowVersion: "0.53"
  propWrapperFunctions:
  - property: freeze
    object: Object
  - property: myFavoriteWrapper
  - property: forbidExtraProps
    exact: true
  componentWrapperFunctions:
  - property: styled
  - property: observer
    object: Mobx
  - property: observer
    object: "<pragma>"
  formComponents:
  - name: Form
    formAttribute: endpoint
  linkComponents:
  - name: Link
    linkAttribute: to
...
```

### ESLint version

8.54.0

### ESLint Plugin Perfectionist version

2.5.0

### Validations

- [X] Read the [docs](https://eslint-plugin-perfectionist.azat.io).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by azat-io** [comment]
There might be a problem with @typescript-eslint/parser. Could you tell me what version you have installed?

And which package manager are you using?

**Comment by evgeniyworkbel** [comment]
> There might be a problem with @typescript-eslint/parser. Could you tell me what version you have installed?
> 
> And which package manager are you using?

Our app is based on create-react-app (v5.0.1, latest). The project does not have typescript-eslint/parser directly, but I assume CRA is used its version 5.57.1. Also yarn v 1.22.19 is installed

![image](https://github.com/azat-io/eslint-plugin-perfectionist/assets/93836126/f7a776e5-6999-43b1-a1a5-d16f935ce051)

**Comment by azat-io** [comment]
Can you create a repository and reproduce the problem?

ESLint Plugin Perfectionist uses TypeScript ESLint v6+.

**Comment by hogan0520** [comment]
Same issue! ESLint 8.15.0, typescript 5

**Comment by azat-io** [comment]
@hogan0520 Can you create a small repository and reproduce the problem?
That would help me a lot to fix the problem.

**Comment by hogan0520** [comment]
> @hogan0520 Can you create a small repository and reproduce the problem? That would help me a lot to fix the problem.

I found the reason. Because `@typescript-eslint` do not have `sourceCode` property in lower version. Just have `getSourceCode()`.
So, when I used `"eslint-plugin-perfectionist": "2.4.0"`, resolved this issue.

Maybe you need to add some code or dependency limitation to handle it. @azat-io 

Thank you for your response.

**Comment by azat-io** [comment]
@hogan0520 Yes, perhaps we should more rigorously specify the version of `@typescript-eslint`. Thank you

**Comment by azat-io** [comment]
Thank you for your issue!

The bug was fixed in 09900aaec7aba76f438c4a910639c3b9fc10832d and released in [v3.0.0](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v3.0.0).

If you'd like, you can support the release with a retweet:
https://x.com/azat_io_en/status/1815367279191761054
